src/commands/lambda/flare.ts

```diff
@@ -150,41 +224,154 @@ export class LambdaFlareCommand extends Command {
+ * Delete a folder and all its contents
+ * @param folderPath the folder to delete
+ * @throws Error if the deletion fails
  */
-export const writeFile = async (folderPath: string, filePath: string, data: string, context: BaseContext) => {
-  if (fs.existsSync(folderPath)) {
-    deleteFolder(folderPath, context)
+export const deleteFolder = (folderPath: string) => {
+  try {
+    fs.rmSync(folderPath, {recursive: true, force: true})
+  } catch (err) {
+    throw Error(`Failed to delete files located at ${folderPath}: ${err.message}`)
   }
+}
 
+/**
+ * Creates the root folder and the logs sub-folder
+ * @param rootFolderPath path to the root folder
+ * @param logsFolderPath path to the logs folder
+ * @param createLogsFolder whether to create the logs folder
+ * @throws Error if the root folder cannot be deleted or folders cannot be created
+ */
+export const createDirectories = (rootFolderPath: string, logsFolderPath: string, createLogsFolder: boolean) => {
+  try {
+    fs.mkdirSync(rootFolderPath)
+    if (createLogsFolder) {
+      fs.mkdirSync(logsFolderPath)
+    }
+  } catch (err) {
+    throw Error(`Unable to create directories: ${err.message}`)
+  }
+}
+
+/**
+ * Gets the 3 latest log stream names, sorted by last event time
+ * @param cwlClient CloudWatch Logs client
+ * @param logGroupName name of the log group
+ * @returns an array of the last 3 log stream names or undefined if no log streams are found
+ * @throws Error if the log streams cannot be retrieved
+ */
+export const getLogStreamNames = async (cwlClient: CloudWatchLogsClient, logGroupName: string) => {
+  const getLogStreamsCommand = new DescribeLogStreamsCommand({
+    logGroupName,
+    limit: 3,
+    descending: true,
+    orderBy: 'LastEventTime',
+  })
+  const logStreams = (await cwlClient.send(getLogStreamsCommand)).logStreams
+  if (logStreams === undefined) {
+    return undefined
+  }
+  const output: string[] = logStreams
+    .filter((logStream) => logStream !== undefined)
+    .map((logStream) => logStream.logStreamName)
+    .filter((logStreamName): logStreamName is string => logStreamName !== undefined && logStreamName.length > 0)
+
+  return output.length === 0 ? undefined : output.reverse()
+  // Reverse array so the oldest log is created first, so Support Staff can sort by creation time
+}
+
+/**
+ * Gets the log events for a log stream
+ * @param cwlClient
+ * @param logGroupName
+ * @param logStreamName
+ * @returns the log events or undefined if no log events are found
+ * @throws Error if the log events cannot be retrieved
+ */
+export const getLogEvents = async (cwlClient: CloudWatchLogsClient, logGroupName: string, logStreamName: string) => {
+  const params = new GetLogEventsCommand({
+    logGroupName,
+    logStreamName,
+  })
+  try {
+    const logEvents = (await cwlClient.send(params)).events
+
+    return logEvents === undefined || logEvents.length === 0 ? undefined : logEvents
+  } catch (err) {
+    throw Error(err.message)
+  }
+}
```

Comment:
```suggestion
export const getLogEvents = async (cwlClient: CloudWatchLogsClient, logGroupName: string, logStreamName: string) => {
  const command = new GetLogEventsCommand({
    logGroupName,
    logStreamName,
  })
  try {
    const response = await cwlClient.send(command)
    const logEvents = response.events

    return logEvents === undefined || logEvents.length === 0 ? undefined : logEvents
  } catch (err) {
    throw err
  }
}
```

Reply:
Thanks, I have made this change.
However, I want to let you know that I still throw using `throw Error(err.message)` to remove the `Error: ` prefix. This is because I catch any errors that `getLogEvents` throws and add an error prefix of `[Error] ` by using `commonRenderer.renderError()`

Reply:
Because I'm throwing an error, it will reappend the `Error: `, so there will be to `Error: ` prefixes.
But I'm realizing that the whole try/catch in `getLogEvents` is unnecessary.

I think it should just be changed to:
```
export const getLogEvents = async (cwlClient: CloudWatchLogsClient, logGroupName: string, logStreamName: string) => {
  const command = new GetLogEventsCommand({
    logGroupName,
    logStreamName,
  })

  try {
    const response = await cwlClient.send(command)
    const logEvents = response.events

    return logEvents === undefined || logEvents.length === 0 ? undefined : logEvents
  } catch (err) {
    throw Error(err.message)
  }
}
```

This way, if `await cliClient.send` throws an error, it will be caught by the caller:
```
          try {
            logEvents = await getLogEvents(cwlClient, logGroupName, logStreamName)
          } catch (err) {
            this.context.stderr.write(
              commonRenderer.renderError(`Unable to get log events for stream ${logStreamName}: ${err.message}`)
            )

            return 1
          }
```

What do you think?